### PR TITLE
Add resbundlers.js that converts json type of resources to js files

### PR DESCRIPTION
**Pull request by gouniLee**
* resbundler is a command-line utility that generates js files from json type of resources

### CODE REVIEW

File: resbundler.js [hunk 1]
```diff
@@ -0,0 +1,147 @@
+/*
```
Comment (ehoogerbeets) [inline resbundler.js:2]: Add this as the first line to get it to execute properly as an executable:

```
#!/usr/bin/env node
```

File: package.json [hunk 2]
```diff
@@ -0,0 +1,27 @@
+{
+  "name": "ilib-resbundler",
+  "version": "1.0.0",
+  "description": "convert json resources to js file",
+  "main": "resbundler.js",
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1"
+  },
```
Comment (ehoogerbeets) [inline package.json:11]: add "dist" target for publishing

File: resbundler.js [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+    if (!fullPath) return;
+    var prefix = "ilib.data.strings";
+
+    var splitPath = fullPath.split("/");
+    var key = splitPath.slice(1, splitPath.length -1).join("_");
+    key = prefix + (key != "" ? "_" + key : key);
+    return key;
+}
+
+function assemblFiles(dir){
```
Comment (ehoogerbeets) [inline resbundler.js]: spelling: `assembleFiles`

File: resbundler.js [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+        }
+    });
+    return result;
+}
+
+function dynamicFiles(){
+    if ((options.assembly != "assembled" || options.assembly == "dynamic") && options.locales.length > 0){
+        options.locales.forEach(function(lo){
+            result = "";
+            lo.split("-").map(function(v, i, a){
```
Comment (ehoogerbeets) [inline resbundler.js]: Use ilib or ilib-locale to parse the locale spec.
Also, better variable names would be good. What do "v", "i" , and "a" mean?

File: resbundler.js [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+                    }
+                }
+            });
+            var filename = lo + ".js";
+            wirteFiles(result, filename);
+        });
+    }
+}
+
+function wirteFiles(writeData, filename){
```
Comment (ehoogerbeets) [inline resbundler.js]: spelling: `writeFiles`

File: package.json (hunk 2 above)
Comment (gouniLee) [reply package.json:11]: I added dist target  and also added test files for checking the result by running npm run test-* command

File: resbundler.js (hunk 4 above)
Comment (gouniLee) [reply resbundler.js]: I updated them. and fixed typo.

File: resbundler.js [hunk 6]
```diff
@@ -107,41 +109,65 @@ function assemblFiles(dir){
...
-                    var data = fs.readFileSync(respath, "utf-8");
-                    var dataKey = manipulateKey(respath);
-                    if (data){
-                        var temp = dataKey +  " =  " + data + ";\n";
-                        result +=temp;
+    if ((options.assembly != "assembled" || options.assembly == "dynamic")){
+        if (options.locales.length > 0) {
+            options.locales.forEach(function(lo){
+                result = "";
+                var loList = parseLocale(lo);
```
Comment (ehoogerbeets) [inline resbundler.js:133]: Take a look at https://github.com/iLib-js/iLib/blob/88b8e6701e5a9b76bb3c46474993065b3d2bbc3d/js/lib/Utils.js#L235
The locales list is a little more complicated than what you have here. Those functions will eventually be in a separate npm package that you can use here.

Comment (ehoogerbeets) [reply resbundler.js:133]: You will do this in a different PR

**Review by ehoogerbeets — APPROVED** [review]

**Pull request merged**